meson.build

```diff
@@ -686,68 +676,58 @@ else
     error('Mandatory header zlib.h not found!')
   endif
   icu_dep = dependency('icu-uc')
-  syslua = get_option('syslua')
-  if syslua != 'false'
-    lua_dep_tmp = dependency('lua-5.4', 'lua-54', 'lua54', 'lua5.4', required:false)
-  endif
 endif
 
 # Set unconditionally, as it was checked as hard requirement
 pub_conf_data.set('FREECIV_HAVE_LIBZ', 1)
 
-mw_req = get_option('mwand')
+mw_dep = dependency('MagickWand', 'MagickWand-6.Q16HDRI', required: get_option('mwand'))
 mw_extra_dep = []
-if mw_req != 'false'
-  mw_dep = dependency('MagickWand', version : '>= 7.0', required : false)
-  if mw_dep.found()
+if mw_dep.found()
+  if mw_dep.version().version_compare('>= 7.0')
     pub_conf_data.set('FREECIV_MWAND7', '1')
     mwand_incl = '#include <MagickWand/MagickWand.h>'
+  elif mw_dep.version().version_compare('>= 6.0')
+    pub_conf_data.set('FREECIV_MWAND6', '1')
+    mwand_incl = '#include <wand/MagickWand.h>'
   else
-    mw_dep = dependency('MagickWand', version : '>= 6.0', required : false)
-    if not mw_dep.found()
-      mw_dep = dependency('MagickWand-6.Q16HDRI', required: false)
-    endif
-    if mw_dep.found()
-      mwand_incl = '#include <wand/MagickWand.h>'
-      mwand_incl = '#include <wand/MagickWand.h>'
-    endif
+    error('MagickWand version 6.0 or newer is required.')
```

Comment:
This is obviously totally bogus. The else branch here does not conditionalize the error on `get_option('mwand').enabled()`. The extra conditionals are the price you pay for not using our (Meson's) builtin functionality.